scripts/ci/check_perl_lsp_rs_core_package.py

```
@@ -0,0 +1,192 @@
+    return args
+
+
+def safe_extract(archive: Path, destination: Path) -> None:
+    destination_resolved = destination.resolve()
+    with tarfile.open(archive, "r:*") as tar:
+        for member in tar.getmembers():
+            target = (destination / member.name).resolve()
+            try:
+                target.relative_to(destination_resolved)
+            except ValueError as error:
+                raise SystemExit(f"refusing to extract path outside destination: {member.name}") from error
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)